packages/edit-navigation/src/components/name-display/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * WordPress dependencies
+ */
+import { ToolbarGroup, ToolbarButton } from '@wordpress/components';
+import { BlockControls } from '@wordpress/block-editor';
+/**
+ * Internal dependencies
+ */
+import {
+	useSelectedMenuData,
+	IsMenuNameControlFocusedContext,
+} from '../../hooks';
+import { useContext } from '@wordpress/element';
+
+import { __ } from '@wordpress/i18n';
+export default function NameDisplay() {
+	const { menuName } = useSelectedMenuData();
+	const [ , setIsMenuNameEditFocused ] = useContext(
+		IsMenuNameControlFocusedContext
+	);
+	return (
+		<BlockControls>
+			<ToolbarGroup>
+				<ToolbarButton
+					aria-label={ __( `Edit menu name:` ) + ' ' + menuName }
```

Comment:
I missed this originally—this will need to use `sprintf` for the string interpolation. There should be plenty of examples across the codebase to follow.